src/pymatgen/core/periodic_table.py

```diff
@@ -1639,6 +1654,7 @@ def get_el_sp(obj: int | SpeciesLike) -> Element | Species | DummySpecies:
             of properties that can be determined.
     """
     # If obj is already an Element or Species, return as is
+    # TODO: Why do we need to check "_is_named_isotope"?
```

Comment:
Please try to avoid inserting a question as a comment for a commit as you're unlikely to get an answer this way.

For a general question, you could use the [discussions section](https://github.com/materialsproject/pymatgen/discussions), for something you suspect to be a bug, please [open an issue](https://github.com/materialsproject/pymatgen/issues)

Perhaps you could open an issue/discussion and discuss there for visibility?

---

I believe this reason is to guard against `DummySpecies` which doesn't have the `_is_named_isotope` attribute, @shyuep in case I'm wrong here. 

```python
from pymatgen.core import DummySpecies

d_sp = DummySpecies("X")
print(d_sp._is_named_isotope)  # AttributeError
```

If I were correct, perhaps this implementation should be tweaked to check type for readability (let's do this in a separate PR):

```diff
- if getattr(obj, "_is_named_isotope", None):
+ if isinstance(obj, DummySpecies):
```

Perhaps there could be exceptions? i.e. a `DummySpecies` with `_is_named_isotope` property?

Reply:
Thanks for writing up the discussion, also link to my previous comment https://github.com/materialsproject/pymatgen/pull/4078#discussion_r1841594048 in case you missed it

Reply:
@esoteric-ephemera do you know anything about the isotope handling here? Was it related to CIF parsing changes?